ldk-server/src/util/config.rs

```diff
@@ -194,79 +113,317 @@ impl Into<LSPS2ServiceConfig> for LSPS2ServiceTomlConfig {
 	}
 }
 
-/// Loads the configuration from a TOML file at the given path.
-pub fn load_config<P: AsRef<Path>>(config_path: P) -> io::Result<Config> {
-	let file_contents = fs::read_to_string(config_path.as_ref()).map_err(|e| {
-		io::Error::new(
-			e.kind(),
-			format!("Failed to read config file '{}': {}", config_path.as_ref().display(), e),
+#[derive(Parser, Debug)]
+#[command(version, about = "LDK Node Configuration", long_about = None)]
+pub struct ArgsConfig {
+	#[arg(required = false)]
+	config_file: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_NODE_NETWORK")]
+	node_network: Option<Network>,
+
+	#[arg(long, env = "LDK_SERVER_NODE_LISTENING_ADDRESS")]
+	node_listening_address: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_NODE_REST_SERVICE_ADDRESS")]
+	node_rest_service_address: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_NODE_ALIAS")]
+	node_alias: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_BITCOIND_RPC_HOST")]
+	bitcoind_rpc_host: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_BITCOIND_RPC_PORT")]
+	bitcoind_rpc_port: Option<u16>,
+
+	#[arg(long, env = "LDK_SERVER_BITCOIND_RPC_USER")]
+	bitcoind_rpc_user: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_BITCOIND_RPC_PASSWORD")]
+	bitcoind_rpc_password: Option<String>,
+
+	#[arg(long, env = "LDK_SERVER_STORAGE_DIR_PATH")]
+	storage_dir_path: Option<String>,
+}
+
+pub fn load_config(args_config: &ArgsConfig) -> io::Result<Config> {
+	let toml_config = if let Some(config_path) = &args_config.config_file {
+		let file_contents = fs::read_to_string(config_path).map_err(|e| {
+			io::Error::new(e.kind(), format!("Failed to read config file '{}': {}", config_path, e))
+		})?;
+
+		Some(toml::from_str::<TomlConfig>(&file_contents).map_err(|e| {
+			io::Error::new(
+				io::ErrorKind::InvalidData,
+				format!("Config file contains invalid TOML format: {}", e),
+			)
+		})?)
+	} else {
+		#[cfg(feature = "events-rabbitmq")]
+		return Err(io::Error::new(
+			io::ErrorKind::InvalidInput,
+			"To use the `events-rabbitmq` feature, you must provide a configuration file.",
+		));
+
+		#[cfg(feature = "experimental-lsps2-support")]
+		return Err(io::Error::new(
+			io::ErrorKind::InvalidInput,
+			"To use the `experimental-lsps2-support` feature, you must provide a configuration file.",
+		));
+
+		None
+	};
+
+	macro_rules! pick {
+		($cli:expr, $toml:expr, $err_msg:expr) => {
+			$cli.or($toml).ok_or_else(|| io::Error::new(io::ErrorKind::InvalidInput, $err_msg))
+		};
+	}
+
+	fn missing_field_msg(field: &str) -> String {
```

Comment:
Could this return the `io::Error` instead? Then the unrolled macro would be less verbose.